crates/hyperswitch_connectors/src/connectors/novalnet.rs

```diff
@@ -719,26 +721,163 @@ impl ConnectorIntegration<Void, PaymentsCancelData, PaymentsResponseData> for No
     }
 }
 
+fn get_webhook_object_from_body(
+    body: &[u8],
+) -> CustomResult<novalnet::NovalnetWebhookNotificationResponse, errors::ConnectorError> {
+    let novalnet_webhook_notification_response = std::str::from_utf8(body)
+        .map_err(|_| errors::ConnectorError::ParsingFailed)
+        .and_then(|body_str| {
+            serde_json::from_str::<novalnet::NovalnetWebhookNotificationResponse>(body_str)
+                .map_err(|_| errors::ConnectorError::WebhookBodyDecodingFailed)
+        })?;
+
+    Ok(novalnet_webhook_notification_response)
+}
+
 #[async_trait::async_trait]
 impl webhooks::IncomingWebhook for Novalnet {
-    fn get_webhook_object_reference_id(
+    fn get_webhook_source_verification_algorithm(
         &self,
         _request: &webhooks::IncomingWebhookRequestDetails<'_>,
+    ) -> CustomResult<Box<dyn crypto::VerifySignature + Send>, errors::ConnectorError> {
+        Ok(Box::new(crypto::Sha256))
+    }
+
+    fn get_webhook_source_verification_signature(
+        &self,
+        request: &webhooks::IncomingWebhookRequestDetails<'_>,
+        _connector_webhook_secrets: &api_models::webhooks::ConnectorWebhookSecrets,
+    ) -> CustomResult<Vec<u8>, errors::ConnectorError> {
+        let notif_item = get_webhook_object_from_body(request.body)
+            .change_context(errors::ConnectorError::WebhookSourceVerificationFailed)?;
+
+        hex::decode(notif_item.event.checksum)
+            .change_context(errors::ConnectorError::WebhookVerificationSecretInvalid)
+    }
+
+    fn get_webhook_source_verification_message(
+        &self,
+        request: &webhooks::IncomingWebhookRequestDetails<'_>,
+        _merchant_id: &common_utils::id_type::MerchantId,
+        connector_webhook_secrets: &api_models::webhooks::ConnectorWebhookSecrets,
+    ) -> CustomResult<Vec<u8>, errors::ConnectorError> {
+        let notif = get_webhook_object_from_body(request.body)
+            .change_context(errors::ConnectorError::WebhookSourceVerificationFailed)?;
+        let (transaction_amount, transaction_currency) = match notif.transaction {
+            novalnet::NovalnetWebhookTransactionData::CaptureTransactionData(data) => {
+                (data.amount, data.currency)
+            }
+            novalnet::NovalnetWebhookTransactionData::CancelTransactionData(data) => {
+                (data.amount, data.currency)
+            }
+
+            novalnet::NovalnetWebhookTransactionData::RefundsTransactionData(data) => {
+                (data.amount, data.currency)
+            }
+
+            novalnet::NovalnetWebhookTransactionData::OtherTransactionData(data) => {
+                (data.amount, data.currency)
+            }
+        };
+        let amount = transaction_amount
+            .map(|amount| amount.to_string())
+            .unwrap_or("".to_string());
```

Comment:
will this lead to source verification failure, in case of empty string ?

Reply:
no "" is intentional, in some events like cancel we wont get currency and amount.